core/arch/arm/kernel/boot.c

```diff
@@ -652,6 +655,10 @@ static int add_dt_overlay_fragment(struct dt_descriptor *dt, int ioffs)
 	int offs;
 	int ret;
 
+#ifdef CFG_OVERLAY_ADDR
+	if (!dt->is_overlay)
+		return ioffs;
```

Comment:
should return an `FDT_ERR_xxx` value, or -1 as done at line 671.
(note -1 is not really nice: it relates to `FDT_ERR_NOTFOUND`. At least it makes the implementation consistent.

Reply:
@etienne-lms um no I dont think so. If this is not the overlay when CFG_OVERLAY_ADDR is enabled, this could behave as if it was a DTB (ie, as if _CFG_USE_DTB_OVERLAY is not defined); and that code return offs.